include/alpaka/core/Assert.hpp

```diff
@@ -89,5 +89,70 @@ namespace alpaka
             ::assertValueUnsigned(
                 arg);
         }
+
+        namespace detail
+        {
+            //#############################################################################
+            template<
+                typename TLhs,
+                typename TRhs,
+                typename TSfinae = void>
+            struct AssertGreaterThan;
+            //#############################################################################
+            template<
+                typename TLhs,
+                typename TRhs>
+            struct AssertGreaterThan<
+                TLhs,
+                TRhs,
+                typename std::enable_if<!std::is_unsigned<TRhs>::value || (TLhs::value != 0u)>::type>
+            {
+                ALPAKA_NO_HOST_ACC_WARNING
+                ALPAKA_FN_HOST_ACC static auto assertGreaterThan(
+                    TRhs const & lhs)
+                -> void
+                {
+#ifdef NDEBUG
+                    alpaka::ignore_unused(lhs);
+#else
+                    assert(TLhs::value > lhs);
+#endif
+                }
+            };
+            //#############################################################################
+            template<
+                typename TLhs,
+                typename TRhs>
+            struct AssertGreaterThan<
+                TLhs,
+                TRhs,
+                typename std::enable_if<std::is_unsigned<TRhs>::value && (TLhs::value == 0u)>::type>
+            {
+                ALPAKA_NO_HOST_ACC_WARNING
+                ALPAKA_FN_HOST_ACC static auto assertGreaterThan(
+                    TRhs const & lhs)
```

Comment:
you can also just remove the `lhs` here instead of using `alpaka::ignore_unused`